internal/server/middleware.go

```diff
@@ -86,6 +92,7 @@ func authenticateRequest(c *gin.Context, tx *data.Transaction, srv *Server) erro
 		return internal.ErrBadRequest
 	}
 
+	// TODO: move to caller
```

Comment:
Now that the transaction is separate, the only thing these two functions provide is the `access.Authenticated` struct. So I think we can return that directly instead of mutating the `gin.Context`.

I'm going to look to address these TODOs in a follow up PR.  It requires a few changes to `handleInfraDestinationHeader`.